src/wombats/lib/wombat_lib.js

```diff
@@ -0,0 +1,282 @@
+// Have module.exports be a function so that user won't have to worry about namespaces
+module.exports = function() {
+  this.turn_directions = ['right', 'left', 'about-face'];
+  this.smoke_directions = ['forward', 'backward', 'left', 'right', 'drop'];
+  this.game_parameters = {
+    // HP Modifiers
+    'collision-hp-damage': 10,
+    'food-hp-bonus': 5,
+    'poison-hp-damage': 10,
+    // Score Modifiers
+    'food-score-bonus': 10,
+    'wombat-hit-bonus': 10,
+    'zakano-hit-bonus': 8,
+    'steel-barrier-hit-bonus': 2,
+    'wood-barrier-hit-bonus': 2,
+    'wombat-destroyed-bonus': 25,
+    'zakano-destroyed-bonus': 15,
+    'wood-barrier-destroyed-bonus': 3,
+    'steel-barrier-destroyed-bonus': 25,
+    // In game parameters
+    'shot-distance': 5
+  };
+  this.shot_range = game_parameters['shot-distance'];
+  this.zakano_key = 'zakano';
+  this.wombat_key = 'wombat';
+  this.steel_key = 'steel-barrier';
+  this.wood_key = 'wood-barrier';
+  this.food_key = 'food';
+  this.poison_key = 'poison';
+  this.fog_key = 'fog';
+  this.open_key = 'open';
+
+  this.point_sources = [zakano_key, wombat_key, steel_key, wood_key, food_key];
+  this.points_no_wall = [zakano_key, wombat_key, food_key];
+  this.enemies = [wombat_key, zakano_key];
+  this.blockers = [zakano_key, wombat_key, wood_key, steel_key, poison_key];
+  this.targets = [zakano_key, wombat_key, steel_key, wood_key];
+  this.persistent = [food_key, poison_key, open_key, wood_key, steel_key];
+
+  this.default_tile = {contents: {type: fog_key}};
+
+  this.mod = function(n, m) {
+    // JS doesn't handle taking the modulus of negative numbers the way python and clojure do
+    return ((n % m) + m) % m;
+  };
+
+  this.get_arena_size = function(state) {
+    return {
+      width: state['global-dimensions'][0],
+      height: state['global-dimensions'][1]
+    };
+  };
+
+  this.add_locs = function(arena) {
+    var arena = JSON.parse(JSON.stringify(arena))
+    for(y = 0; y < arena.length; y++) {
+      var row = arena[y];
+      for(x = 0; x < row.length; x++) {
+        arena[y][x].x = x;
+        arena[y][x].y = y;
+      };
+    };
+    return arena;
+  };
+
+  this.filter_arena = function(arena, filters = null) {
+    if (filters == null) {
+      return [].concat.apply([], arena);
+    } else {
+      var flattened = [].concat.apply([], arena);
+      return flattened.filter(function(tile) {
```

Comment:
[Arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) will make this cleaner.